src/components/chat/SendMessage.tsx

```diff
@@ -0,0 +1,117 @@
+import { FC, RefObject } from 'react';
+import { FlatList, StyleSheet } from 'react-native';
+import { replaceMentionValues } from 'react-native-controlled-mentions';
+import { IMessage, Send } from 'react-native-gifted-chat';
+
+import { MaterialIcons } from '@expo/vector-icons';
+
+import { UUID } from '@graasp/sdk';
+
+import {
+  MENTION_CHAT_TRIGGER,
+  PRIMARY_COLOR,
+  PRIMARY_LIGHT_COLOR,
+} from '../../config/constants/constants';
+import { ChatMessage } from '../../config/types';
+import { useQueryClient } from '../../context/QueryClientContext';
+import { convertIMessageToText } from '../../utils/functions/chat';
+
+interface SendMessageProps {
+  itemId: UUID;
+  text?: string;
+  messageSelected: ChatMessage | null;
+  chatRef?: RefObject<FlatList<IMessage>>;
+  handleMessageSelected: (message: ChatMessage | null) => void;
+  handleIsEditMessage: (value: boolean) => void;
+  handleInputMessage: (inputText: string) => void;
+}
+
+const SendMessage: FC<SendMessageProps> = ({
+  itemId,
+  text,
+  messageSelected,
+  chatRef,
+  handleMessageSelected,
+  handleIsEditMessage,
+  handleInputMessage,
+}) => {
+  const { mutations } = useQueryClient();
+  const { mutate: postMessage } = mutations.usePostItemChatMessage();
+  const { mutate: patchMessage } = mutations.usePatchItemChatMessage();
+
+  const postChatMessage = async (
+    newMessageText: string,
+    mentions?: string[],
+  ) => {
+    postMessage({
+      itemId,
+      body: newMessageText,
+      mentions: mentions || [],
+    });
+  };
+
+  const patchChatMessage = async (editedMessageText: string) => {
+    if (messageSelected) {
+      patchMessage({
+        itemId,
+        messageId: messageSelected._id as string,
```

Comment:
Did you add this `_id` yourself? I think this needs a custom type to avoid casting. Like `ChatMessageWithInternalId`?